File: WorldFeed/Server/src/Services/Trend/WorldFeed.Trend.API/Client/Validators/TrendsClientRequiredParametersValidator.cs
namespace WorldFeed.Trend.Client.Validators
{
    using System;

    using WorldFeed.Trend.Application.Parameters;

    public interface ITrendsClientRequiredParametersValidator : ITrendsClientParametersValidator
    {
    }

    public class TrendsClientRequiredParametersValidator : ITrendsClientRequiredParametersValidator
    {
        public void Validate(IGetTrendsLocationCloseToParameters parameters)
        {
            if (parameters == null)
            {
                throw new ArgumentNullException(nameof(parameters));
            }
        }

        public void Validate(IGetTrendsAtParameters parameters)
        {
            if (parameters == null)
            {
                throw new ArgumentNullException(nameof(parameters));
            }
        }

        public void Validate(IGetTrendsLocationParameters parameters)
        {
            if (parameters == null)
            {
                throw new ArgumentNullException(nameof(parameters));
            }
        }
    }
}

File: Sharebook/Client/src/app/core/Core/Client/Validators/TrendsClientRequiredParametersValidator.ts
import {Injectable, InjectionToken} from "@angular/core";

import {ITrendsClientParametersValidator} from "./TrendsClientParametersValidator";
import {TrendsParameters} from "./parameters-types";
import ArgumentNullException from "typescript-dotnet-commonjs/System/Exceptions/ArgumentNullException";

export interface ITrendsClientRequiredParametersValidator extends ITrendsClientParametersValidator {
}

export const ITrendsClientRequiredParametersValidatorToken = new InjectionToken<ITrendsClientRequiredParametersValidator>('ITrendsClientRequiredParametersValidator', {
  providedIn: 'root',
  factory: () => new TrendsClientRequiredParametersValidator(),
});

@Injectable({
  providedIn: 'root',
})
export class TrendsClientRequiredParametersValidator implements ITrendsClientRequiredParametersValidator {
  public validate(parameters: TrendsParameters): void {
    if (parameters == null) {
      throw new ArgumentNullException(`nameof(parameters)`);
    }
  }
}

